File: nomad-samples/portal/py/guest_registration/account/participant_panel_query.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json
import requests

# @param {string} AUTH_TOKEN - The authentication token

def participant_panel_query(AUTH_TOKEN: str, ID: str) -> dict:
    if not AUTH_TOKEN:
        raise Exception("Authentication token not found")

    API_URL = "https://" + PORTAL_API_URL + "/user-session/" + ID

    HEADERS = {
        "Content-Type": "application/json",
      	"Authorization": "Bearer " + AUTH_TOKEN
    }
    
    BODY = {
        "id": ID
    }
    
    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        if not RESPONSE.ok:
            raise Exception()

        return json.loads(RESPONSE.text)
    except:
      raise Exception("Participant panel query filed")

File: nomad-samples/portal/js/guest-registration/guest/participant-panel-query.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js"; 

export default async function participantPanelQuery(AUTH_TOKEN, ID) 
{
		// Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
  	const BODY = {
        id: ID
    };
  
  // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/user-session/${ID}`, {
        method: "POST",
        mode: "no-cors",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Participant panel query failed");
}